uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -1770,72 +1804,96 @@ or a phantom
+        $this->writeErrorLog($message, $NeuroDB::ExitCodes::INSERT_FAILURE);
+        $this->spool($message, 'Y', $upload_id, $notify_notsummary);
+
         $CandMismatchError = 'CandID does not exist';
+
         return $CandMismatchError;
     }
-    
+
+    print "Candidate ID $candID found\n" if ($this->{verbose});
+
+
     #################################################################
     ### Check if PSCID exists and that PSCID and CandID of the scan
     # refers to the same candidate
     #################################################################
     $query = "SELECT CandID, PSCID FROM candidate WHERE PSCID=?";
-    $sth =  ${$this->{'dbhr'}}->prepare($query);
-    $sth->execute($subjectIDsref->{'PSCID'});
+    $sth   = $dbh->prepare($query);
+    $sth->execute($pscID);
+
     if ($sth->rows == 0) {
-        print LOG "\n=> No PSCID\n";
-        $CandMismatchError= 'PSCID does not exist';
+        $message = "\n\n=> Could not find candidate with PSCID=$pscID in database\n";
+        $this->writeErrorLog($message, $NeuroDB::ExitCodes::INSERT_FAILURE);
+        $this->spool($message, 'Y', $upload_id, $notify_notsummary);
+
+        $CandMismatchError = 'PSCID does not exist';
+
         return $CandMismatchError;
     } else {
         # Check that the PSCID and CandID refers to the same candidate
         my $rowref = $sth->fetchrow_hashref;
-        unless ($rowref->{'CandID'} == $subjectIDsref->{'CandID'}) {
-            $CandMismatchError = 'PSCID and CandID of the image mismatch\n';
-            print LOG "\n=> $CandMismatchError";
+        unless ($rowref->{'CandID'} == $candID) {
+            $CandMismatchError = "PSCID and CandID of the image mismatch\n";
+            $this->writeErrorLog($CandMismatchError, $NeuroDB::ExitCodes::INSERT_FAILURE);
+            $this->spool($CandMismatchError, 'Y', $upload_id, $notify_notsummary);
+
             return $CandMismatchError;
         }
     }
 
-    ############################################################
-    ################ No Checking if the subject is Phantom #####
-    ############################################################
-    if ($subjectIDsref->{'isPhantom'}) {
-        # CandID/PSCID errors don't apply to phantoms, so we don't
-        # want to trigger
-        # the check which aborts the insertion
-        $CandMismatchError = undef;
-        return $CandMismatchError;
-    }
 
     ############################################################
-    ################ Check if visitLabel exists ################
+    ################ Check if visit label is valid##############
     ############################################################
-
     $query = "SELECT Visit_label FROM Visit_Windows WHERE BINARY Visit_label=?";
-    $sth =  ${$this->{'dbhr'}}->prepare($query);
-    $sth->execute($subjectIDsref->{'visitLabel'});
+    $sth   = ${$this->{'dbhr'}}->prepare($query);
+    $sth->execute($visit_label);
+
```

Comment:
Instead of creating a statement and then executing it, you can do this all at once:

```perl
my @rows = $this->{'dbhr'}->selectall_array($query, {}, $visit_label);
```

and then check if `@rows == 0` afterwards. I think there are other occurence of this pattern elsewhere in the code also.

Reply:
done